api/src/main/java/org/apache/iceberg/expressions/Predicate.java

```diff
@@ -19,21 +19,24 @@
 
 package org.apache.iceberg.expressions;
 
-public abstract class Predicate<R extends Reference> implements Expression {
+import com.google.common.base.Preconditions;
+
+public abstract class Predicate<T, C extends Term> implements Expression {
   private final Operation op;
-  private final R ref;
+  private final C child;
 
-  Predicate(Operation op, R ref) {
+  Predicate(Operation op, C child) {
+    Preconditions.checkNotNull(child, "Term cannot be null");
     this.op = op;
-    this.ref = ref;
+    this.child = child;
   }
 
   @Override
   public Operation op() {
     return op;
   }
 
-  public R ref() {
-    return ref;
+  public C child() {
```

Comment:
This may be somewhat pedantic, but the name `child` doesn't seem right here.  I would suggest `comparand`, which is used in other literature.

Reply:
I updated it to `term`.